game/src/js/util/Controller.js

```diff
@@ -1,19 +1,53 @@
 export default class Controller {
   constructor(scene) {
     this.cursors = scene.input.keyboard.createCursorKeys();
-
+    this.keyW = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.W);
+    this.keyA = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.A);
+    this.keyS = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.S);
+    this.keyD = scene.input.keyboard.addKey(Phaser.Input.Keyboard.KeyCodes.D);
     
   }
 
   getWASDCoordinate() {
     let direction = this.cursors.down.isDown ? 'S' : (this.cursors.up.isDown ? 'N' : '');
     direction += this.cursors.left.isDown ? 'W' : (this.cursors.right.isDown ? 'E' : '');
+
+    direction += this.keyA.isDown ? 'W' : (this.keyD.isDown ? 'E' : '');
```

Comment:
If both the left arrow key and A are pressed at the same time, the final direction would be 'WW'. Will rarely happen, but when it does lots of errors will occur.

Maybe something like `direction += this.cursors.left.isDown || this.keyW.isDown ? 'W' : (this.cursors.right.isDown || this.keyD.isDown = 'E' : '')` will take care of this problem.